presto-main/src/main/java/com/facebook/presto/sql/planner/plan/SemiJoinNode.java

```diff
@@ -38,6 +38,7 @@
     private final Optional<Symbol> sourceHashSymbol;
     private final Optional<Symbol> filteringSourceHashSymbol;
     private final Optional<DistributionType> distributionType;
+    private final boolean legacySemiJoin;
```

Comment:
I think the field name should suggest how the operator works, e.g: produces values in binary or ternary logic. So the name could be: `semiJoinOutputNullDistinguishable` or `semiJoinOutputInTernaryLogic`, etc.